### DISCUSSION TITLE
Expose a way to insert RPU (and maybe ITU_T_T35) in HEVC bitstream

### DISCUSSION THREAD

**Discussion by galad87** [quietvoid/dovi_tool#245] [Ideas]
If I understand correctly, dovi_tool cli can already insert those. I wonder if it would be doable to expose these feature in libdovi thru a C API. It would be useful for example to expand HDR10+ and Dolby Vision support in HandBrake to more encoders (only a few encoders supports RPU and ITU_T_T35), I would avoid rewriting things when there is already a good working solution.

**Comment by quietvoid** [comment]
It would probably be quite a bit of effort and requires #155 to be done first.

**Comment by quietvoid** [comment]
@galad87 Quick question on this.
Since the operation could take a while, should the API work by async callbacks to call when finished?
I assume it should as to not block, but I don't know if work is already separate from UI in HandBrake.

**Comment by galad87** [comment]
HandBrake is already multi-threaded, each part of the pipeline (decoder/filters/encoder/muxer) runs on a different thread, so in its case an async callback is not required.

**Reply by quietvoid** [reply]
Okay. I guess it would be easier for it to be blocking, then.
